src/main/java/org/mule/module/apikit/validation/attributes/HeadersValidator.java

```diff
@@ -206,4 +210,33 @@ private static List<String> getResponseMimeTypes(Map<String, Response> responses
     return new ArrayList<>();
   }
 
+
+  public static boolean someRepresentationIsSupported(List<String> supportedRepresentations, String header) {
+    List<MimeType> parseResults;
+    try {
+      parseResults = MimeType.listFrom(header, COMMA.getDelimiterChar());
+    } catch (MimeType.MimeTypeParseException e) {
+      LOGGER.warn("Failed to parse user-provided MimeType list: {}", header, e);
+      return false;
+    }
+
+    return supportedRepresentations.stream()
+        .flatMap(representation -> {
+          try {
+            return Stream.of(MimeType.from(representation));
+          } catch (MimeType.MimeTypeParseException e) {
+            LOGGER.warn("Failed to parse application-provided MimeType: {}", representation, e);
+            return Stream.empty();
+          }
+        })
+        .anyMatch(supported -> parseResults.stream().anyMatch(expected -> isCompatible(expected, supported)));
+  }
+
+  private static boolean isCompatible(MimeType expected, MimeType supported) {
+    return isCompatible(expected.getType(), supported.getType()) && isCompatible(expected.getType(), supported.getType());
+  }
+
+  private static boolean isCompatible(String expected, String supported) {
+    return expected.equalsIgnoreCase(supported) || "*".equals(expected);
```

Comment:
There's a behavior change here. Previously we were saying that if the user requests `image/png` and we support `image/*` then that's a match.

The issue is that supporting `image/*` doesn't really make sense (because that means "I know how to generate any kind of image, even ones yet to be developed").